Feature/my/vocal-analysis-report

dona0123 opened the pull request:
## 🛠️ 작업내용
- VocalAnalysisReport (보컬 분석 리포트) 화면 구현
- 게스트 보컬 분석 리포트 생성 api 연결 

## 📜 이슈번호
#34 

## 📢 전달사항
- "나의 음역대"는 인간의 전체 음역대에서 차지하는 비율로 계산했습니다. 
- "완성도"는 나머지 4개 항목(음정 박자, 호흡, 발음)의 평균값으로 지정했습니다.  
- 여러 리포트 저장, analysis_type에 따른 UI 변경은 추후 구현 예정입니다. 
<img src="https://github.com/user-attachments/assets/a5070dd6-be02-4136-908d-fb212e64393b" width="400" />

hj9023 committed:
✨ Feat: 마이크 퍼미션 획득 및 기본 오디오 로직 구현 #25

hj9023 committed:
♻️ Refactor: 오디오 녹음 관련 로직 개선

hj9023 committed:
♻️ Refactor: 오디오 녹음 관련 로직 개선 #25

hj9023 committed:
✨ Feat: 오디오 녹음 로직 구현 #25

hj9023 committed:
🧹 Chore: 브랜치 병합 오류 해결:  'origin/feature/pitch-detection'와  'feature/pitch-detection' 브랜치

20211234 committed:
🧹 Chore: 음정 감지 라이브러리 실험

dona0123 committed:
✨ Feat: 보컬 분석 리포트 화면 구현 #34

dona0123 committed:
🧹 Chore: 임시 데이터 수정 #34

hj9023 committed:
♻️ Refactor: 음정 분석 로직을 provider/controller 구조로 리팩토링하여 재사용성 개선 #25

20211234 committed:
✨ Feat: 음정 기능 개발

20211234 committed:
🐞 fix: 라이브러리 교체 (ffmpeg_kit → record)

hj9023 committed:
✨ Feat: 실시간 음정 탐지 기능 구현 #25

hj9023 committed:
🧹 Chore: feature/pitch-detection 과 feature/pitch-detection-test/eunsun 브랜치 병합 #25

hj9023 committed:
✨ Feat: .wav확장자로 녹음 파일 저장 #25

hj9023 committed:
✨ Feat: .wav확장자로 녹음 파일 저장 #25

hj9023 committed:
🧹 Chore: 브랜치 feature/pitch-detection-test/merge-branch 병합 완료 #25

20211234 committed:
🧹 Chore: 테스트 음원(=도레미송) 추가

20211234 committed:
✨ Feat: 음정 막대 임시 구현

hj9023 committed:
✨ Feat: 평균/최저/최고 음정 분석 및 저장 #38
- 평균/최저/최고 페이지 실시간 음성 받기
- 주파수 50-2000 사이의 값만 음정으로 받는 로직 추가

hj9023 commented on lib/features/my/presentation/pages/vocal_analysis_report_page.dart:
```diff
@@ -1,11 +1,588 @@
+  String get voiceType =>
+      _getVoiceTypeKorean(songData['voice_type'] as String? ?? '');
+  String get voiceRange => songData['voice_range'] as String? ?? '0~0';
+  List<String> get voiceRangeParts => voiceRange.split('~');
```
사용자의 음역대를 표시하는 거 맞죠?? 그럼 프론트에서 저장한 거 쓰는 거라서 제가 지금 구현하는 부분인데, 아마 프로바이더에서 꺼내 쓰는 식으로 사용하시면 될 것 같습니다.

hj9023 commented on lib/features/my/presentation/pages/vocal_analysis_report_page.dart:
```diff
@@ -1,11 +1,588 @@
+
+    final userRange = endValue - startValue;
+    return (userRange / totalHumanRange).clamp(0.0, 1.0);
+  }
```
이것도 사용자 최저/최고 음정 맞죠? 최저/최고 음정도 아마 위의 음역대 정보랑 같이 주게 될 것 같습니다!!  프로바이더에서 값 가져가는 식으로 가져가게 만들고 있습니다. (ref.watch(voiceTypeProfileProvider).minNote / .maxNote . / .voiceType 이런 식으로..)

20211234 reviewed (approved):
🍧

hj9023 committed:
✨ Feat: 최저/최고 노트 및 음역대 정보 저장 #38
- 불필요한 패키지 삭제
- 프로바이더 변수 이름 변경 : pitchStatsProvider -> vocalPItchMetricsProvider

20211234 committed:
🐞 Fix: 음정 막대 높낮이 오류 해결

hj9023 committed:
Merge remote-tracking branch 'origin/develop' into feature/pitch-detection-session-logic

dona0123 committed:
🧹 Chore: 데이터 형식 수정 #34

hj9023 committed:
🐞 Fix: 음정 탐지 오류 및 재생/중지 버튼 클릭 지연 문제 해결
- audio_stream_recorder 의 pitch 감지 수정: pitched == false 인 경우에 가짜 데이터를 보냄

hj9023 committed:
🧹 Chore: 브랜치 머지 - fature/sing_bar 와 feature/pitch-detection-session-logic

hj9023 committed:
🐞 Fix: 머지 후 발생한 유효하지 않은 pitch 데이터를 사용하는 부분 수정

hj9023 committed:
✨ Feat: 음성 파일 경로 및 정확도 저장을 위한 상태 관리 로직 추가

hj9023 committed:
✨ Feat: 음정 정확도 분석 구현

hj9023 committed:
♻️ Refactor: evaluatePitch 로직 개선 및 정확도 계산 구조 분리

- bool 기반 pitch 평가 → int 음정 차이 기반으로 변경
- pitch 정확도 계산을 별도 함수로 분리
- evaluatedPitchStreamProvider를 별도 파일로 분리하여 역할 명확화

20211234 committed:
♻️ Refactor: 새로운 doremi json 파일 추가 및 연결

hj9023 committed:
♻️ Refactor: 성능이 개선된 doremi json 파일 추가 및 recordingSongPage 퍼미션 확인 코드 추가

dona0123 committed:
Merge branch 'feature/scoring-pitch' of github.com:SoLit-2025/sync2sing-fe into feature/my/vocal-analysis-report

dona0123 committed:
✨ Feat: 보컬 분석 리포트로 이동 경로 수정 #34

dona0123 committed:
✨ Feat: 보컬 분석 리포트 api 요청 #34

dona0123 committed:
✨ Feat: api 응답 저장 #34

20211234 committed:
💄 Design: 가사 크기 축소 및 정렬 수정

20211234 committed:
♻️ Refactor: 정확도 100% 음정 막대 데이터(.json) 및 MR 음원 추가

20211234 committed:
🧹 Chore: assets/audios 폴더 내 모든 파일을 프로젝트에 사용할 수 있도록 설정 수정

20211234 committed:
✨ Feat: 실시간 박자 채점 기능 추가

dona0123 committed:
✨ Feat: 보컬 분석 리포트 생성 api 연결 #34

hj9023 committed:
🐞 Fix: 박자 정확도 저장 오류 문제 해결
- 박자 정확도 저장 위치 수정
- 현재 음정 바 위치를 프로바이더에서 알 수 있게 수정

hj9023 committed:
🐞 Fix: doremi 가사를 원본에 맞게 수정 및 불필요한 json 파일 삭제

dona0123 committed:
🐞 Fix: 사용자의 음역대와 성부를 출력하도록 수정 #34

dona0123 committed:
Merge branch 'feature/scoring-rhythm-0614' of github.com:SoLit-2025/sync2sing-fe into feature/my/vocal-analysis-report

# Conflicts:
#	lib/features/training_common/presentation/controller/audio_recorder_controller.dart

dona0123 committed:
🔥 Remove: 사용하지 않는 파일 삭제 #34

dona0123 committed:
🎨 Style: 주석 정리 및 불필요한 gitkeep 삭제 #34
